rest-api-spec/schema.json

```diff
@@ -0,0 +1,242 @@
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "deprecated": {
+                    "$ref": "#/definitions/Deprecated"
+                },
+                "documentation": {
+                    "$ref": "#/definitions/Documentation"
+                },
+                "stability": {
+                    "type": "string",
+                    "enum": ["stable", "beta", "experimental", "private"]
+                },
+                "url": {
+                    "$ref": "#/definitions/Url"
+                },
+                "params": {
+                    "$ref": "#/definitions/Params"
+                },
+                "body": {
+                    "$ref": "#/definitions/Body"
+                }
+            },
+            "required": [
+                "documentation",
+                "stability",
+                "url"
+            ],
+            "title": "API components",
+            "description": "The components that make up the API"
+        },
+        "Url": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "paths": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Path"
+                    },
+                    "minLength": 1
+                }
+            },
+            "required": [
+                "paths"
+            ],
+            "title": "API urls",
+            "description": "An array of the URL path variants for the API"
+        },
+        "Params": {
+            "type": "object",
+            "additionalProperties": true,
+            "propertyNames": {
+              "pattern": "^_?[a-z]+?(_[a-z]+)*$"
+            },
+            "patternProperties": {
+                "^_?[a-z]+?(_[a-z]+)*$": {
+                    "$ref": "#/definitions/ParamPart"
+                }
+            },
+            "title": "API query string parameters",
+            "description": "A collection of the query string parameters for the API"
+        },
+        "Body": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "description": {
+                    "type": "string"
+                },
+                "required": {
+                    "type": "boolean"
+                },
+                "serialize": {
+                    "type": "string",
+                    "enum": ["bulk"]
+                }
+            },
+            "required": [
+                "description"
+            ],
+            "title": "API request Body",
+            "description": "The request body for the API. Does not detail the structure, only whether the API accepts a body and its format"
+        },
+        "Deprecated": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "version": {
```

Comment:
should we expand this to list to allow saying this was deprecrated in 6.8.7 and 7.6.1 ? (probably not with this PR, but as a follow up)

Reply:
Happy to do in a follow up. I don’t think we have a need for this in the current specs, but do you see one coming up?

Reply:
Not specifically, and now that I think about this abit more it is likely pretty rare for a need to do this.

Reply:
No worries - can always amend down the line